administration/src/bp-modules/user-settings/ApiTokenSettings.tsx

```diff
@@ -0,0 +1,147 @@
+import { Button, H2, H4, HTMLSelect, HTMLTable } from '@blueprintjs/core'
+import Delete from '@mui/icons-material/Delete'
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+
+import getMessageFromApolloError from '../../errors/getMessageFromApolloError'
+import {
+  ApiTokenMetaData,
+  useCreateApiTokenMutation,
+  useDeleteApiTokenMutation,
+  useGetApiTokenMetaDataQuery,
+} from '../../generated/graphql'
+import { formatDate } from '../../util/formatDate'
+import { useAppToaster } from '../AppToaster'
+import getQueryResult from '../util/getQueryResult'
+import SettingsCard from './SettingsCard'
+
+const Container = styled.div`
+  background: ghostwhite;
+  padding: 20px;
+  border-radius: 8px;
+  box-shadow: inset 0 2px 4px rgba(0, 0, 0, 0.05);
+`
+
+const Row = styled.div`
+  width: 80%;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  margin-bottom: 15px;
+`
+
+const NewTokenText = styled.p`
+  font-size: 18px;
+  padding: 10px;
+  border: 1px solid black;
+  border-radius: 6px;
+  margin-top: 15px;
+  word-break: break-all;
+`
+
+const ApiTokenSetting = () => {
+  const metaDataQuery = useGetApiTokenMetaDataQuery({})
+
+  const appToaster = useAppToaster()
+
+  const [tokenMetaData, setTokenMetadata] = useState<Array<ApiTokenMetaData>>([])
+  const [createdToken, setCreatedToken] = useState<string>('')
+  const [expiresIn, setExpiresIn] = useState<number>(1)
+
+  useEffect(() => {
+    const metaDataQueryResult = getQueryResult(metaDataQuery)
+    if (metaDataQueryResult.successful) {
+      const { tokenMetaData } = metaDataQueryResult.data
+      setTokenMetadata(tokenMetaData)
+    }
+  }, [metaDataQuery])
+
+  const [createToken] = useCreateApiTokenMutation({
+    onCompleted: result => {
+      appToaster?.show({ intent: 'success', message: 'Token wurde erfolgreich erzeugt.' })
+      setCreatedToken(result.createApiTokenPayload)
+      metaDataQuery.refetch()
+    },
+    onError: error => {
+      const { title } = getMessageFromApolloError(error)
+      appToaster?.show({
+        intent: 'danger',
+        message: title,
+      })
+    },
+  })
+
+  const [deleteToken] = useDeleteApiTokenMutation({
```

Comment:
optional: maybe would be nice to add a confirmation dialog? to avoid accidental deletion